### PR TITLE
Add Go wrappers for MBA<2>

### PR DESCRIPTION
**Pull request by yonch**
Thank you for open-sourcing this implementation! It had been a pleasure to work with on a Python project.

I wanted to use MBA<2> in Golang, so wrote these wrappers using `cgo`...

Verified using tests (in `golang_mba_test.go`):
* a port of the "small test function" test in `test_mba.cpp`
* generated several values using the `interp` Python example in the README, the Golang test then expects to generate the exact same values for the same inputs.

This implementation also support the functionality through Python's `__repr__` via the `String()` function on the Mba2 struct.

I hope to continue contributing further fixes or enhancements in case those come up.

Thanks again for open sourcing!

### CODE REVIEW

**Comment by ddemidov** [comment]
Thank you for the contribution! Does it need any changes in the CMakeLists.txt to build the new test?

**Comment by yonch** [comment]
I test manually by running `go test` in the `mba` folder, but it will be good to add to the CMakeLists.txt if it's part of your workflow -- I'll go ahead and add that

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,15 @@ RELEASE-VERSION
...
+CMakeCache.txt
+CMakeFiles
+CMakeScripts
+Testing
+Makefile
+cmake_install.cmake
+install_manifest.txt
+compile_commands.json
+CTestTestfile.cmake
+_deps
```
Comment (ddemidov) [inline .gitignore:19]: These should not be a problem with out-of-source builds, but ok

**Comment by ddemidov** [comment]
Thanks again!

File: .gitignore (hunk 1 above)
Comment (yonch) [reply .gitignore:19]: agreed, I usually build out of build too. fixed it on autopilot..

**Comment by yonch** [comment]
thank you for the quick turnaround!! hope to contribute more in the future :)

**Comment by ddemidov** [comment]
I have applied fixes for the go tests on travis-ci in 15cbf8de28de6a32f9b2f28568758032dd91d7a6 (the logs are [here](https://travis-ci.org/github/ddemidov/mba/jobs/774531405)). Hope the fixes make sense.

**Pull request merged**